### DISCUSSION TITLE
Option to download 720p videos instead.

### DISCUSSION THREAD

**Discussion by ImSoDan** [UltimaHoarder/UltimaScraper#676] [Ideas]
As you may know, OnlyFans re-encodes videos uploaded to their platform, and they seem to do a terrible job.  If the actual source video is 60fps, the 'source' video is still re-encoded to 30 or 24fps.  However, the 720p video usually has the original source framerate or at least is above 50fps, and thus usually looks much better.  Having the option to choose the video quality would be a great addition because of this.  I apologize if this already exists and I'm missing something.

**Comment by James-MF-Thomas** [comment]
To my knowledge I’ve only ever seen the source footage be downloaded. Unless you want the degraded vids m, it’s not something to worry about

**Reply by ImSoDan** [reply]
Hey James,

It does tell you that it's the 'source', but if you compare the two files you can tell immediately that the 'lower' quality file is actually the same exact bitrate, and a higher framerate, if the uploaded source was in fact higher framerate to begin with.  The fact is that the 'source' file has been modified by OnlyFans and has ahd frames removed.  While the end resolution is lower in the 720p files, in my opinion the higher frame rate is perceived as higher quality.

**Comment by UltimaHoarder** [comment]
Thank you, I didn't know this. I will add this option soon 👍

**Comment by James-MF-Thomas** [comment]
frame rate does not necessarily correlate to quality. That being said 24,30,60 is def better than 7 or 15. Unless you know what the original frame rate was when it was shot. Footage shot at 24fps then converted to 120fps, Isn't going to look any better its just going to be sped up. Footage shot at 120fps per and converted to 30fps, isn't going to look any better, its just going to be in slow motion. 

Being that OF re-encodes the videos, a standard practice for web encoding to reduce size is to lower the bit-rate. After that to go a step further in reducing size and still maintaining a respectable amount of quality in the image s to reduce the frame rate. less frames, less bytes.  the lower bit rate is usually a dead give a way for lesser quality, even though it may not be noticeable visually.

I'm not disagreeing with you, if it looks like shitty encoding, and smells like shitty encoding, its prob shitty encoding. Though you can't judge its lossless-ness from frame-rate alone. but good looking out comparing the images samples. I just took the scrapes from here at face value. I know if you pull the vids yourself the "source" label should the best of the versions.

**Reply by ImSoDan** [reply]
What you are saying would be true if the 'source' videos provided by OF were actually the source videos, however they are not.  I wish they would offer a 1080p file that actually included the true source framerate, but sadly they dont.  Obviously re-encoding at a higher framerate that the source isn't better, but the reverse is happening. They are taking a higher framerate source and cutting it in half.  Of course the way I realized this in the first place was by noticing the smoother playback and investigating further.

**Comment by avekifes** [comment]
I was wondering why the videos I had downloaded before this script came into existence seemed higher in quality. I thought that OnlyFans had simply compressed everything to save on server space.

**Reply by ImSoDan** [reply]
It's disappointing honestly. The best source would surely be the full 1080p videos with the source framerate intact, but it seems like they don't want to exceed 6000kbps.

**Comment by UltimaHoarder** [comment]
The latest commit has the feature.

240p,720p, and source are supported.

If they ever add in 1080p, it'd automatically be supported as well.

If the 720p videos aren't downloading, you'll need to set overwrite files to true.

**Comment by headphonia** [comment]
While I agree that 720p is generally better than source, there seems to be 2 exceptions.

1. When the model uploads something in 4K it seems to remain in 4K as the "source".
2. 1080p files uploaded around April/May 2020 seem to be actual 1080p rather than transcoded/compressed/upscaled 1080p

That being said, I'm not a coder, so I don't know if there's an easy way to determine which file to scrape without comparing them manually.

**Reply by nappyapp** [reply]
@DIGITALCRIMINAL is it possible for the script to determine this difference just by downloading headers or would it need to download both files to check? would be cool if this could be automatically detected

**Comment by hsg120** [comment]
I see exactly what OP is talking about. I requested content in 60fps, but it only shows the higher framerate when I select the 720p resolution. Whyyyyyyyy OnlyFans............

To that end--I couldn't get the 720p option to work in the config file either. It continues to download the "source" version (at 30fps). I tried changing the 'overwrite files' option also to no avail.

**Comment by porncritic** [comment]
I use an extension with dl option source or 720p, usually I dl both to check, and most of the time source is better quality. At least on my computer, with a big difference usually
